HIRS_Utils/src/main/java/hirs/tpm/eventlog/uefi/UefiGuid.java

```diff
@@ -0,0 +1,415 @@
+ case "c1c41626-504c-4092-aca9-41f936934328":reference = "EFI_CERT_SHA256_GUID"; break;
+ case "826ca512-cf10-4ac9-b187-be01496631bd":reference = "EFI_CERT_SHA1_GUID"; break;
+ case "67f8444f-8743-48f1-a328-1eaab8736080":reference = "EFI_CERT_RSA2048_SHA1_GUID"; break;
+ case "a5c059a1-94e4-4aa7-87b5-ab155c2bf072":reference = "EFI_CERT_X509_GUID"; break;
+ case "0b6e5233-a65c-44c9-9407-d9ab83bfc8bd":reference = "EFI_CERT_SHA224_GUID"; break;
+ case "ff3e5307-9fd0-48c9-85f1-8ad56c701e01":reference = "EFI_CERT_SHA384_GUID"; break;
+ case "093e0fae-a6c4-4f50-9f1b-d41e2b89c19a":reference = "EFI_CERT_SHA512_GUID"; break;
+ case "3bd2a492-96c0-4079-b420-fcf98ef103ed":reference = "EFI_CERT_X509_SHA256_GUID"; break;
+ case "7076876e-80c2-4ee6-aad2-28b349a6865b":reference = "EFI_CERT_X509_SHA384_GUID"; break;
+ case "446dbf63-2502-4cda-bcfa-2465d2b0fe9d":reference = "EFI_CERT_X509_SHA512_GUID"; break;
+ case "a7717414-c616-4977-9420-844712a735bf":
+                reference = "EFI_CERT_TYPE_RSA2048_SHA256_GUID"; break;
+ // UEFI defined variables
+ case "452e8ced-dfff-4b8c-ae01-5118862e682c":
+                reference = "EFI_CERT_EXTERNAL_MANAGEMENT_GUID"; break;
+ case "d719b2cb-3d3a-4596-a3bc-dad00e67656f":reference = "EFI_IMAGE_SECURITY_DATABASE_GUID"; break;
+ case "4aafd29d-68df-49ee-8aa9-347d375665a7":reference = "EFI_CERT_TYPE_PKCS7_GUID"; break;
+ case "c12a7328-f81f-11d2-ba4b-00a0c93ec93b" :reference = "EFI System Partition"; break;
+ case "024DEE41-33E7-11D3-9D69-0008C781F39F" :
+                reference = "Partition containing a legacy MBR"; break;
+ // RHBoot UEFI Application UUIDs
+ // From listed in RHBoot (RHShim) https://github.com/rhboot/efivar/blob/master/src/guids.txt
+ case "0abba7dc-e516-4167-bbf5-4d9d1c739416":reference = "fwupdate:"; break;
+ case "3b8c8162-188c-46a4-aec9-be43f1d65697":reference = "ux_capsule"; break;
+ case "605dab50-e046-4300-abb6-3dd810dd8b23":reference = "RH_Shim"; break;
+ case "8be4df61-93ca-11d2-aa0d-00e098032b8c":reference = "EFI_Global_Variable"; break;
+ case "91376aff-cba6-42be-949d-06fde81128e8":reference = "GRUB"; break;
+
+ // Partition Table GUIDs
+ case "0fc63daf-8483-4772-8e79-3d69d8477de4":reference = "Linux filesystem data"; break;
+ case "e6d6d379-f507-44c2-a23c-238f2a3df928":
+                reference = "Logical Volume Manager (LVM) partition"; break;
+ case "4f68bce3-e8cd-4db1-96e7-fbcaf984b709":reference = "Root partition (x86-64)"; break;
+ case "a19d880f-05fc-4d3b-a006-743f0f84911e":reference = "RAID partition "; break;
+ case "933ac7e1-2eb4-4f13-b844-0e14e2aef915":reference = "/home partition[ (x86-64)"; break;
+ case "ebd0a0a2-b9e5-4433-87c0-68b6b72699c7":reference = "GPT Basic data partition"; break;
+
+ // RHBoot Lenovo specific UUIDS
+ case "3cc24e96-22c7-41d8-8863-8e39dcdcc2cf":reference = "lenovo"; break;
+ case "82988420-7467-4490-9059-feb448dd1963":reference = "lenovo_me_config"; break;
+ case "f7e615b-0d45-4f80-88dc-26b234958560":reference = "lenovo_diag"; break;
```

Comment:
This has a different length.  Is this the correct full ID?

Reply:
Wow, that's a good catch. It's missing a leading 3.